speaker/layer2_controller_test.go

```diff
@@ -222,10 +224,42 @@ func TestUsableNodesEPSlices(t *testing.T) {
 			usableSpeakers:  map[string]bool{"iris1": true, "iris2": true},
 			cExpectedResult: []string{"iris1", "iris2"},
 		},
+		{
+			desc:                     "Two endpoints, different hosts, ready but not serving and publishNotReadyAddresses is true",
+			publishNotReadyAddresses: true,
+			eps: []discovery.EndpointSlice{
+				{
+					Endpoints: []discovery.Endpoint{
+						{
+							Addresses: []string{
+								"2.3.4.5",
+							},
+							NodeName: ptr.To("iris1"),
+							Conditions: discovery.EndpointConditions{
+								Ready:   ptr.To(true),
+								Serving: ptr.To(true),
```

Comment:
All of the endpoint's serving is true, It looks like your tests and descriptions are consistent.